src/docfx/build/moniker/MonikersProvider.cs

```diff
@@ -10,6 +11,7 @@ namespace Microsoft.Docs.Build
     internal class MonikersProvider
     {
         private readonly List<(Func<string, bool> glob, string monikerRange)> _rules = new List<(Func<string, bool>, string)>();
+        private readonly ConcurrentDictionary<Document, List<string>> _monikersMap = new ConcurrentDictionary<Document, List<string>>();
```

Comment:
Should it be `ConcurrentDictionary<Document, Lazy<List<string>>>` and use `GetOrAdd` as atomic action?

Reply:
we shouldn't mantains the map here, will move it to build.cs

---
In reply to: [235276852](https://github.com/dotnet/docfx/pull/3655#discussion_r235276852) [](ancestors = 235276852)